File: 프로그래머스/1/135808. 과일 장수/과일 장수.py
def solution(k, m, score):
    answer = 0
    
    score.sort()
    eachBoxMin = []
    count = 0
    min = 9
    
    for i in range(len(score)-1,-1,-1):
        if score[i] <= min:
            min = score[i]
            count += 1
            if count == m:
                eachBoxMin.append(min)
                count = 0
    for i in eachBoxMin:
        answer += i*m
    
    return answer

File: 프로그래머스/1/135808. 과일 장수/과일 장수.java
import java.util.*;

class Solution {
    public int solution(int k, int m, int[] score) {
        int answer = 0;
        ArrayList<Integer> arr = new ArrayList<>();
        
        Arrays.sort(score);
        
        for(int i = score.length - 1; i >= 0; i--){
            if ((score.length - i) % m == 0) {
                arr.add(score[i]);
            }
        }
        
        for(int a : arr) {
            answer += a * m;
        }
        
        return answer;
    }
}